flowetl/etl/tests/test_construct_etl_dag.py

```diff
@@ -35,8 +39,18 @@ def test_construct_etl_dag_with_test_callables():
 
     assert dag.dag_id == f"etl_{cdr_type}"
 
-    dag_task_callable_mapping = {t.task_id: t.python_callable for t in dag.tasks}
-    assert dag_task_callable_mapping == TEST_ETL_TASK_CALLABLES
+    dag_task_callable_mapping = {
+        t.task_id: t.python_callable
+        for t in dag.tasks
+        if not isinstance(t, PostgresOperator)
```

Comment:
Why is this check needed? Won't all tasks used in the test run be instances of `PythonOperator`?

(If we do need to filter them, might be useful to be explicit that we require them to be python operators: `if isinstance(t, PythonOperator)`?)

Reply:
They will all be `PythonOperator` in this specific scenario, but the one below will have the "production" ones; these tests check that they are properly set for a DAG - this was needed for the second one so it seemed consistent to have both of them like that. Although, they will be `PythonOperator` or `BranchPythonOperator`, but it makes sense to make it inclusive rather than exclusive. Updated the checks to check for instances of `PythonOperator` and `BranchPythonOperator`.

Reply:
I just realised that `BranchPythonOperator` is actually a subclass of `PythonOperator` so we only need a check for `isinstance(..., PythonOperator)`.

Reply:
Because the user-defined macros don't render as templates, it wasn't as simple to write one as `"extract_table": "etl.x{{ cdr_type }}_{{ ds_nodash }}"`. Instead, they have been made into lambdas and pass the `ds_nodash` through the templates (`"extract_table": lambda execution_time: f"etl.x{ cdr_type }_{ execution_time }",` and `{{ extract_table(ds_nodash) }}`).

What do you think of that? It can be either that, or making the extract_table as an extract_table_prefix instead and using `{{ extract_table_prefix }}_{{ ds_nodash }}` in the templates, OR passing the entire dag_run instead of ds_nodash and using dag_run.conf inside the lambda.